### DISCUSSION TITLE
Ökofen JSON Compatibility Mode: Values appear with factor 10x

### DISCUSSION THREAD

**Discussion by Jens314** [dominikamann/oekofen-pellematic-compact#13] [General]
Hi Dominik, thanks for this wonderful Integration. I have just started playing around and getting into all of this home assistant stuff. Anyway I managed to get things running and I can see the sensor values in HA, see picture. Unfortunately by factor 10x too high.
As I can see in your code at some point you are dividing the values by 10.
Do you have an idea what is going wrong here?
Thanks Jens
 
![grafik](https://user-images.githubusercontent.com/125146110/218270320-adc9455c-40ba-4885-ba7c-569904a7267b.png)

**Comment by dominikamann** [comment]
Hi @Jens314, 
that's strange. For me and other user it works fine. Can you please provide me your JSON File? Just open your heater URL in a browser, copy the content and attach it here. Then iam able to investigate...
Thanks!

**Comment by Jens314** [comment]
sure, here it is:

{
   "system":{
      "L_ambient":"82",
      "L_errors":"0",
      "L_usb_stick":"false"
   },
   "weather":{
      "L_temp":"40",
      "L_clouds":"100",
      "L_forecast_temp":"60",
      "L_forecast_clouds":"67",
      "L_forecast_today":"false",
      "L_starttime":"920",
      "L_endtime":"1550",
      "L_source":"http://www.openweathermap.org",
      "L_location":"xxxx",
      "cloud_limit":"55",
      "hysteresys":"-40",
      "offtemp":"-100",
      "lead":"120",
      "refresh":"false",
      "oekomode":"0"
   },
   "forecast":{
      "L_w_0":"Sa, 11 Feb 20:06|4|100|1 km/h|04n|804|C|07:37|17:34",
      "L_w_1":"Sa, 11 Feb 19:00|4|88|1 km/h|04n|804|C",
      "L_w_2":"Sa, 11 Feb 22:00|2|92|1 km/h|04n|804|C",
      "L_w_3":"So, 12 Feb 01:00|1|95|1 km/h|04n|804|C",
      "L_w_4":"So, 12 Feb 04:00|1|91|2 km/h|04n|804|C",
      "L_w_5":"So, 12 Feb 07:00|2|94|4 km/h|04n|804|C",
      "L_w_6":"So, 12 Feb 10:00|3|100|4 km/h|04d|804|C",
      "L_w_7":"So, 12 Feb 13:00|6|98|8 km/h|04d|804|C",
      "L_w_8":"So, 12 Feb 16:00|7|37|10 km/h|03d|802|C",
      "L_w_9":"So, 12 Feb 19:00|3|20|7 km/h|02n|801|C",
      "L_w_10":"So, 12 Feb 22:00|1|0|6 km/h|01n|800|C",
      "L_w_11":"Mo, 13 Feb 01:00|0|0|5 km/h|01n|800|C",
      "L_w_12":"Mo, 13 Feb 04:00|0|0|5 km/h|01n|800|C",
      "L_w_13":"Mo, 13 Feb 07:00|0|0|6 km/h|01n|800|C",
      "L_w_14":"Mo, 13 Feb 10:00|4|0|6 km/h|01d|800|C",
      "L_w_15":"Mo, 13 Feb 13:00|8|0|7 km/h|01d|800|C",
      "L_w_16":"Mo, 13 Feb 16:00|7|0|7 km/h|01d|800|C",
      "L_w_17":"Mo, 13 Feb 19:00|3|0|6 km/h|01n|800|C",
      "L_w_18":"Mo, 13 Feb 22:00|1|0|6 km/h|01n|800|C",
      "L_w_19":"Di, 14 Feb 01:00|1|0|5 km/h|01n|800|C",
      "L_w_20":"Di, 14 Feb 04:00|0|0|5 km/h|01n|800|C",
      "L_w_21":"Di, 14 Feb 07:00|0|0|5 km/h|01n|800|C",
      "L_w_22":"Di, 14 Feb 10:00|5|0|3 km/h|01d|800|C",
      "L_w_23":"Di, 14 Feb 13:00|9|0|4 km/h|01d|800|C",
      "L_w_24":"Di, 14 Feb 16:00|9|0|5 km/h|01d|800|C"
   },
   "hk1":{
      "L_roomtemp_act":"0",
      "L_roomtemp_set":"220",
      "L_flowtemp_act":"290",
      "L_flowtemp_set":"296",
      "L_comfort":"0",
      "L_state":"32",
      "L_statetext":"Heizbetrieb aktiv",
      "L_pump":"true",
      "remote_override":"0",
      "mode_auto":"1",
      "time_prg":"0",
      "temp_setback":"180",
      "temp_heat":"220",
      "temp_vacation":"150",
      "name":"Fu??bodenHZG",
      "oekomode":"0",
      "autocomfort":"-1",
      "autocomfort_sunset":"0",
      "autocomfort_sunrise":"0"
   },
   "hk2":{
      "L_roomtemp_act":"0",
      "L_roomtemp_set":"220",
      "L_flowtemp_act":"352",
      "L_flowtemp_set":"411",
      "L_comfort":"0",
      "L_state":"32",
      "L_statetext":"Heizbetrieb aktiv",
      "L_pump":"true",
      "remote_override":"0",
      "mode_auto":"1",
      "time_prg":"0",
      "temp_setback":"180",
      "temp_heat":"220",
      "temp_vacation":"150",
      "name":"Heizk??rper",
      "oekomode":"0",
      "autocomfort":"-1",
      "autocomfort_sunset":"0",
      "autocomfort_sunrise":"0"
   },
   "thirdparty1":{
      "L_state":"1|SHHT-1#6A5DB3#|SHHT-1#6A5DB3#|false|0|0|0|1676127686629|192.168.178.74"
   },
   "pu1":{
      "L_tpo_act":"595",
      "L_tpo_set":"461",
      "L_tpm_act":"404",
      "L_tpm_set":"461",
      "L_pump_release":"80",
      "L_pump":"0",
      "L_state":"512",
      "L_statetext":"Anforderung Aus",
      "mintemp_off":"80",
      "mintemp_on":"80",
      "ext_mintemp_off":"80",
      "ext_mintemp_on":"80"
   },
   "ww1":{
      "L_temp_set":"420",
      "L_ontemp_act":"591",
      "L_offtemp_act":"595",
      "L_pump":"false",
      "L_state":"8208",
      "L_statetext":"Zeit innerhalb Zeitprogramm|Anforderung Aus",
      "time_prg":"0",
      "sensor_on":"0",
      "sensor_off":"1",
      "mode_auto":"1",
      "mode_dhw":"1",
      "heat_once":"false",
      "temp_min_set":"400",
      "temp_max_set":"500",
      "name":"",
      "smartstart":"0",
      "use_boiler_heat":"0",
      "oekomode":"0"
   },
   "circ1":{
      "L_pummp":"false",
      "L_ret_temp":"334",
      "L_release_temp":"591",
      "time_prg":"0",
      "mode":"1",
      "pump_release":"400",
      "return_set":"370",
      "name":""
   },
   "pe1":{
      "L_temp_act":"654",
      "L_temp_set":"80",
      "L_ext_temp":"-32768",
      "L_frt_temp_act":"786",
      "L_frt_temp_set":"80",
      "L_br":"false",
      "L_ak":"false",
      "L_not":"true",
      "L_stb":"true",
      "L_modulation":"0",
      "L_uw_speed":"0",
      "L_state":"99",
      "L_statetext":"Aus",
      "L_type":"1",
      "L_starts":"1477",
      "L_runtime":"2323",
      "L_avg_runtime":"94",
      "L_uw_release":"600",
      "L_uw":"0",
      "L_storage_fill":"0",
      "L_storage_min":"400",
      "L_storage_max":"6000",
      "L_storage_popper":"0",
      "storage_fill_today":"0",
      "storage_fill_yesterday":"0",
      "mode":"1"
   },
   "error":{
   }
}

**Comment by lucaferrucci** [comment]
Hello, good morning, the same thing happened to me, in the pellematic when you activate json you don't have to set compatible mode, after which you will see the correct values

**Comment by Jens314** [comment]
Thanks Luca, I can remember that I activated this "compatibility mode" accidentally. But now I can not get back to this option in the menu!? Not by deactivating JSON and not even by switching the complete heater off/on. How did you change it back?

**Comment by lucaferrucci** [comment]
you must enter the code number hour plus date day


> Il giorno 13 feb 2023, alle ore 21:30, Jens314 ***@***.***> ha scritto:
> 
> Thanks Luca, I can remember that I activated this "compatibility mode" accidentally. But now I can not get back to this option in the menu!? Not by deactivating JSON and not even by switching the complete heater off/on. How did you change it back?
> 
>

**Comment by Jens314** [comment]
ok, thanks, I tried this too. Still no possibility to change the "compatibility mode". A technician told me on the phone this may be a bug caused by an old SW version. Well, maybe I can find a solution around this.

**Comment by lucaferrucci** [comment]
what software version do you have?

**Comment by Jens314** [comment]
Touch V3.10d

**Comment by lucaferrucci** [comment]
hello, I absolutely recommend you to upgrade to 4.01

**Comment by dominikamann** [comment]
@Jens314 the issue seems to be that in the compatibility mode the values are all "Strings" and not "Numbers". I will check if I can find a workaround.

**Comment by dominikamann** [comment]
@Jens314 please try my new hot fix "v2.0.2". I updated also the read me that the compatibility mode is not supported/recommended. Please let me know if it is working now for you ...

**Reply by Jens314** [reply]
Thanks so much :-D it is working now!

**Reply by dominikamann** [reply]
@Jens314 why converting to float? I think all the values are Integers in the JSON. So I converting "615" (String) to 615 (Integer). That's the hotfix. Then dividing 615 / 10 = 61,5 Degrees as before.

**Comment by Jens314** [comment]
Hab den Kommentar schon gelöscht, funktioniert auch so. Danke nochmal

Dominik Amann ***@***.***> schrieb am Mi. 15. Feb. 2023 um
19:52:

**Reply by dominikamann** [reply]
Ah perfekt. Dann ist das Thema ja erledigt. Falls es noch was gibt einfach melden 👍
